### PR TITLE
Build Redis from source.

### PR DESCRIPTION
**Pull request by udan11**
* Official redis package ([redis-server](https://packages.ubuntu.com/xenial/misc/redis-server)) is 3.0.6.
* By building from source we have a better control over the version.
* These changes are upgrading Redis to 5.0.3.

### CODE REVIEW

File: image/base/install-redis [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+chmod 750 /var/log/redis
+
+# Install and enabling service.
+cat <<EOF > /etc/systemd/system/redis.service
+[Unit]
+Description=Advanced key-value store
+After=network.target
+Documentation=http://redis.io/documentation, man:redis-server(1)
+
+[Service]
```
Comment (SamSaffron) [inline image/base/install-redis]: I don't think we need any of this stuff ... with use runit to spawn it all we need is the binary.

**Comment by SamSaffron** [comment]
I am merging this, can you build a new base image asap @udan11 and confirm a clean install works with it?

File: image/base/Dockerfile [hunk 2]
```diff
@@ -128,7 +131,7 @@ ADD boot /sbin/boot
 ADD cron /etc/service/cron/run
 ADD rsyslog /etc/service/rsyslog/run
 ADD cron.d_anacron /etc/cron.d/anacron
-
+ADD redis /etc/service/redis/run
```
Comment (xfalcox) [inline image/base/Dockerfile]: Remove this line

File: image/base/redis [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (xfalcox) [inline image/base/redis]: Remove this file.

Move contents to https://github.com/discourse/discourse_docker/blob/master/templates/redis.template.yml#L2-L8

**Pull request merged**